types/src/ledger/receipt.rs

```diff
@@ -28,9 +29,12 @@ impl Receipt {
     }
 
     // Generates a proof for the hash of the log at a given index.
-    pub fn generate_proof_for_log_hash(&self, log_index: usize) -> Result<MerkleProof> {
+    pub fn generate_proof_for_log_hash(
+        &self,
+        log_index: usize,
+    ) -> Result<MerkleProof, MerkleError> {
```

Comment:
This should also be an option it should either have such a log or not be possible to fail